pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/ConsumerConfig.java

```diff
@@ -0,0 +1,111 @@
+import com.google.pubsub.kafkastubs.common.config.ConfigDef.Type;
+import com.google.pubsub.kafkastubs.common.serialization.Deserializer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * The consumer configuration keys
+ */
+class ConsumerConfig extends AbstractConfig {
+  private static final ConfigDef CONFIG;
+
+    /*
+     * NOTE: DO NOT CHANGE EITHER CONFIG STRINGS OR THEIR JAVA VARIABLE NAMES AS
+     * THESE ARE PART OF THE PUBLIC API AND CHANGE WILL BREAK USER CODE.
+     */
+  /** <code>max.poll.records</code> */
+  static final String MAX_POLL_RECORDS_CONFIG = "max.poll.records";
+  private static final String MAX_POLL_RECORDS_DOC =
+      "The maximum number of records returned in a single call to poll() FOR SINGLE TOPIC.";
+
+  /** <code>key.deserializer</code> */
+  static final String KEY_DESERIALIZER_CLASS_CONFIG = "key.deserializer";
+  private static final String KEY_DESERIALIZER_CLASS_DOC =
+      "Deserializer class for key that implements the <code>Deserializer</code> interface.";
+
+  /** <code>value.deserializer</code> */
+  static final String VALUE_DESERIALIZER_CLASS_CONFIG = "value.deserializer";
+  private static final String VALUE_DESERIALIZER_CLASS_DOC =
+      "Deserializer class for value that implements the <code>Deserializer</code> interface.";
+
+  static {
+    CONFIG = new ConfigDef()
+        .define(MAX_POLL_RECORDS_CONFIG,
+            Type.INT,
+            Importance.MEDIUM,
+            MAX_POLL_RECORDS_DOC)
+        .define(KEY_DESERIALIZER_CLASS_CONFIG,
+            Type.CLASS,
+            Importance.HIGH,
+            KEY_DESERIALIZER_CLASS_DOC)
+        .define(VALUE_DESERIALIZER_CLASS_CONFIG,
+            Type.CLASS,
+            Importance.HIGH,
+            VALUE_DESERIALIZER_CLASS_DOC);
+
+  }
+
+  static Map<String, Object> addDeserializerToConfig(Map<String, Object> configs,
+      Deserializer<?> keyDeserializer,
+      Deserializer<?> valueDeserializer) {
+    Map<String, Object> newConfigs = new HashMap<String, Object>();
+    newConfigs.putAll(configs);
+    if (keyDeserializer != null)
+      newConfigs.put(KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer.getClass());
+    if (valueDeserializer != null)
+      newConfigs.put(VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer.getClass());
+    return newConfigs;
+  }
+
+  static Properties addDeserializerToConfig(Properties properties,
+      Deserializer<?> keyDeserializer,
+      Deserializer<?> valueDeserializer) {
+    Properties newProperties = new Properties();
+    newProperties.putAll(properties);
+    if (keyDeserializer != null)
+      newProperties.put(KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer.getClass().getName());
+    if (valueDeserializer != null)
+      newProperties.put(VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer.getClass().getName());
+    return newProperties;
+  }
+
+  ConsumerConfig(Map<?, ?> props) {
```

Comment:
Multiple constructors are not preferred in Java. Consider using a static factory as suggested in Effective Java Item 1